File: Latihan/LatihanIdlingResource/java/app/src/main/java/com/dicoding/picodiploma/myidleresource/MainActivity.java
package com.dicoding.picodiploma.myidleresource;

import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.dicoding.picodiploma.myidleresource.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding activityMainBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        activityMainBinding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(activityMainBinding.getRoot());

        activityMainBinding.button.setOnClickListener(v -> {
            delay1();
            delay2();
        });
    }

    private void delay1() {
        EspressoIdlingResource.increment();
        new Handler().postDelayed(() -> {
            activityMainBinding.textView.setText(getString(R.string.delay1));

            if (!EspressoIdlingResource.getEspressoIdlingResource().isIdleNow()) {
                //Memberitahukan bahwa tugas sudah selesai dijalankan
                EspressoIdlingResource.decrement();
            }

        }, 2000);
    }

    private void delay2() {
        EspressoIdlingResource.increment();
        new Handler().postDelayed(() -> {
            activityMainBinding.textView.setText(getString(R.string.delay2));

            if (!EspressoIdlingResource.getEspressoIdlingResource().isIdleNow()) {
                //Memberitahukan bahwa tugas sudah selesai dijalankan
                EspressoIdlingResource.decrement();
            }
        }, 3000);
    }
}

File: Latihan/LatihanIdlingResource/kotlin/app/src/main/java/com/dicoding/picodiploma/myidleresource/MainActivity.kt
package com.dicoding.picodiploma.myidleresource

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.appcompat.app.AppCompatActivity
import com.dicoding.picodiploma.myidleresource.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var activityMainBinding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        activityMainBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(activityMainBinding.root)

        activityMainBinding.button.setOnClickListener {
            delay1()
            delay2()
        }
    }

    private fun delay1() {
        EspressoIdlingResource.increment()
        Handler(Looper.getMainLooper()).postDelayed({
            activityMainBinding.textView.text = getString(R.string.delay1)
            if (!EspressoIdlingResource.getEspressoIdlingResource().isIdleNow) {
                //Memberitahukan bahwa tugas sudah selesai dijalankan
                EspressoIdlingResource.decrement()
            }
        }, 2000)
    }

    private fun delay2() {
        EspressoIdlingResource.increment()
        Handler(Looper.getMainLooper()).postDelayed({
            activityMainBinding.textView.text = getString(R.string.delay2)
            if (!EspressoIdlingResource.getEspressoIdlingResource().isIdleNow) {
                //Memberitahukan bahwa tugas sudah selesai dijalankan
                EspressoIdlingResource.decrement()
            }
        }, 3000)
    }
}